### PR TITLE
gun system(Almost done just wanna make sure its all good with master branch)

### PR DESCRIPTION
**Pull request by dark00shadow**
hello, mmm i finsihed for gun1 kinda

### CODE REVIEW

File: src/externalEvents/gunsystem.json [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+      "type": "BuiltinCommonInstructions::Comment",
+      "color": {
+        "b": 109,
+        "g": 230,
+        "r": 255,
+        "textB": 0,
+        "textG": 0,
+        "textR": 0
+      },
+      "comment": "gun stuff",
```
Comment (arthuro555) [inline src/externalEvents/gunsystem.json:126]: Maybe some more straightforward comments?

File: src/externalEvents/niko.json [hunk 2]
```diff
@@ -164,72 +164,6 @@
       "source": "",
       "type": "BuiltinCommonInstructions::Group",
       "events": [
-        {
```
Comment (arthuro555) [inline src/externalEvents/niko.json:167]: Why is niko loosing functionality?

File: src/game.json.autosave [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "macExecutableFilename": "",
```
Comment (arthuro555) [inline src/game.json.autosave:14]: Please remove the autosave ;)

File: src/layouts/overworld.json [hunk 4]
```diff
@@ -22,7 +22,7 @@
     "gridWidth": 70,
     "snap": true,
     "windowMask": false,
-    "zoomFactor": 0.5
+    "zoomFactor": 0.418
```
Comment (arthuro555) [inline src/layouts/overworld.json:25]: why did you change the zoom factor?

File: src/layouts/rooms.json [hunk 5]
```diff
@@ -5493,25 +5493,6 @@
               "target": "Niko"
             }
           ]
-        },
-        {
```
Comment (arthuro555) [inline src/layouts/rooms.json:5497]: Why did that event get removed?

File: src/layouts/overworld.json (hunk 4 above)
Comment (arthuro555) [reply src/layouts/overworld.json:25]: nvm I confused that value with another

File: src/externalEvents/gunsystem.json (hunk 1 above)
Comment (dark00shadow) [reply src/externalEvents/gunsystem.json:126]: okay by gun a stuff i mean drop and shooting

File: src/externalEvents/niko.json (hunk 2 above)
Comment (dark00shadow) [reply src/externalEvents/niko.json:167]: huh

File: src/game.json.autosave (hunk 3 above)
Comment (dark00shadow) [reply src/game.json.autosave:14]: huh

File: src/layouts/rooms.json (hunk 5 above)
Comment (dark00shadow) [reply src/layouts/rooms.json:5497]: idk

File: src/externalEvents/niko.json (hunk 2 above)
Comment (arthuro555) [reply src/externalEvents/niko.json:167]: Events from niko have been deleted

File: src/layouts/rooms.json (hunk 5 above)
Comment (dark00shadow) [reply src/layouts/rooms.json:5497]: idk i didnt mess with rooms

File: src/game.json.autosave (hunk 3 above)
Comment (dark00shadow) [reply src/game.json.autosave:14]: kay

File: src/externalEvents/gunsystem.json (hunk 1 above)
Comment (arthuro555) [reply src/externalEvents/gunsystem.json:126]: It's not me who needs to understand, but whoever else will read the code. The comment have to be straightforward in the code, or not included.

File: src/externalEvents/niko.json (hunk 2 above)
Comment (dark00shadow) [reply src/externalEvents/niko.json:167]: idk all i changed was a little in that ooh and the animation becuase i didnt see the point of it and anothe reasons

File: src/externalEvents/gunsystem.json (hunk 1 above)
Comment (dark00shadow) [reply src/externalEvents/gunsystem.json:126]: okay

**Comment by dark00shadow** [comment]
changed it now how do i do a requst for it now lol
